app/delegate-server/Contract/QueryAuction.purs

```diff
@@ -0,0 +1,97 @@
+module DelegateServer.Contract.QueryAuction
+  ( queryAuction
+  ) where
+
+import Contract.Prelude
+
+import Contract.Monad (Contract)
+import Contract.Transaction (TransactionInput)
+import Contract.Utxos (getUtxo)
+import Control.Error.Util ((!?), (??))
+import Control.Monad.Except (ExceptT, throwError, withExceptT)
+import Data.Validation.Semigroup (validation)
+import HydraAuctionOffchain.Contract.Types
+  ( class ToContractError
+  , AuctionInfoExtended
+  , AuctionInfoValidationError
+  , AuctionTermsValidationError
+  , ContractOutput
+  , mkAuctionInfoExtended
+  , mkContractOutput
+  , validateAuctionInfo
+  , validateAuctionTerms
+  )
+import HydraAuctionOffchain.Contract.Validators (MkAuctionValidatorsError, mkAuctionValidators)
+import HydraAuctionOffchain.Helpers (getInlineDatum)
+
+queryAuction :: TransactionInput -> Contract (ContractOutput AuctionInfoExtended)
```

Comment:
Is it L2 specific?

Reply:
No, but it's only used in app/delegate-server.